src/core/qgismobileapp.cpp

```diff
@@ -912,9 +923,90 @@ void QgisMobileapp::print( int layoutIndex )
+  QgsLayoutExporter::ExportResult result = exporter.exportToPdf( destination, pdfSettings );
+
+  if ( result == QgsLayoutExporter::Success )
+    PlatformUtilities::instance()->open( destination );
+
+  return result == QgsLayoutExporter::Success ? true : false;
+}
+
+bool QgisMobileapp::printAtlasFeatures( const QString &layoutName, const QList<long long> &featureIds )
+{
+  const QList<QgsPrintLayout *> printLayouts = mProject->layoutManager()->printLayouts();
+  QgsPrintLayout *layoutToPrint = nullptr;
+  for( QgsPrintLayout *layout : printLayouts )
+  {
+    if ( layout->name() == layoutName )
+    {
+      layoutToPrint = layout;
+      break;
+    }
+  }
+
+  if ( !layoutToPrint || !layoutToPrint->atlas() )
+    return false;
+
+  QStringList ids;
+  for( const auto id : featureIds )
+  {
+    ids << QString::number( id );
+  }
+
+  QString error;
+  layoutToPrint->atlas()->setFilterExpression( QStringLiteral( "$id IN (%1)" ).arg( ids.join( ',' ) ), error );
+  layoutToPrint->atlas()->setFilterFeatures( true );
+
+  const QString documentsLocation = QStringLiteral( "%1/QField" ).arg( QStandardPaths::writableLocation( QStandardPaths::DocumentsLocation ) );
+  QDir documentsDir( documentsLocation );
+  if ( !documentsDir.exists() )
+    documentsDir.mkpath( "." );
+  const QString destination = documentsLocation + '/' + layoutToPrint->name() + QStringLiteral( ".pdf" );
+
+  QgsLayoutExporter::PdfExportSettings pdfSettings;
+  pdfSettings.rasterizeWholeImage = layoutToPrint->customProperty( QStringLiteral( "rasterize" ), false ).toBool();
+  pdfSettings.dpi = layoutToPrint->renderContext().dpi();
+  pdfSettings.appendGeoreference = true;
+  pdfSettings.exportMetadata = true;
+  pdfSettings.simplifyGeometries = true;
+
+  QVector< double > mapScales = layoutToPrint->project()->viewSettings()->mapScales();
+  bool hasProjectScales( layoutToPrint->project()->viewSettings()->useProjectScales() );
+  if ( !hasProjectScales || mapScales.isEmpty() )
+  {
+    // default to global map tool scales
+    const QStringList scales = Qgis::defaultProjectScales().split( ',' );
+    for ( const QString &scale : scales )
+    {
+      QStringList parts( scale.split( ':' ) );
+      if ( parts.size() == 2 )
+      {
+        mapScales.push_back( parts[1].toDouble() );
+      }
+    }
+  }
+  pdfSettings.predefinedMapScales = mapScales;
+
+  if ( layoutToPrint->atlas()->updateFeatures() )
+  {
+    QgsLayoutExporter exporter = QgsLayoutExporter( layoutToPrint );
+    QgsLayoutExporter::ExportResult result;
+    if ( layoutToPrint->customProperty( QStringLiteral( "singleFile" ), true ).toBool() )
+    {
+      result = exporter.exportToPdf( layoutToPrint->atlas(), destination, pdfSettings, error );
+      if ( result == QgsLayoutExporter::Success )
+        PlatformUtilities::instance()->open( destination );
+    } else {
+      result = exporter.exportToPdfs( layoutToPrint->atlas(), destination, pdfSettings, error );
+#ifndef Q_OS_ANDROID
```

Comment:
So it should open every single file on "not android"?

Reply:
No, on windows/linux, it opens the container folder.

Reply:
Ah perfrct